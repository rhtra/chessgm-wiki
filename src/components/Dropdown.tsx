type Option = {
  value: string;
  label: string;
};

type Props = {
  label: string;
  value: string;
  options: Option[];
  onChange: (value: string) => void;
  order?: "asc" | "desc";
  onOrderChange?: (order: "asc" | "desc") => void;
};

export default function Dropdown({
  label,
  value,
  options,
  onChange,
  order,
  onOrderChange,
}: Props) {
  return (
    <div className="dropdown">
      <label className="dropdown-label">{label}:</label>
      <div className="dropdown-control">
        <select
          className="dropdown-select"
          value={value}
          onChange={(e) => onChange(e.target.value)}
        >
          {options.map((opt) => (
            <option key={opt.value} value={opt.value}>
              {opt.label}
            </option>
          ))}
        </select>
        {order && onOrderChange && (
          <button
            type="button"
            className="order-btn"
            onClick={() => onOrderChange(order === "asc" ? "desc" : "asc")}
          >
            {order === "asc" ? "↑" : "↓"}
          </button>
        )}
      </div>
    </div>
  );
}
