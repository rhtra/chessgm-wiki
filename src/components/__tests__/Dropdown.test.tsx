import { render } from "@testing-library/react";
import Dropdown from "../Dropdown";

const options = [
  { value: "username", label: "Username" },
  { value: "firstName", label: "First Name" },
  { value: "lastName", label: "Last Name" },
];

describe("Dropdown snapshots", () => {
  it("matches snapshot (default)", () => {
    const { asFragment } = render(
      <Dropdown label="Sort by" value="username" options={options} onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with order toggle", () => {
    const { asFragment } = render(
      <Dropdown
        label="Sort by"
        value="firstName"
        options={options}
        onChange={() => {}}
        order="asc"
        onOrderChange={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
