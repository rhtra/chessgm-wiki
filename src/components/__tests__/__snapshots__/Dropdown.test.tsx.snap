// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dropdown snapshots matches snapshot (default) 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <label
      class="dropdown-label"
      for="dropdown-sort-by"
    >
      Sort by:
    </label>
    <div
      class="dropdown-control"
    >
      <select
        class="dropdown-select"
        id="dropdown-sort-by"
      >
        <option
          value="username"
        >
          Username
        </option>
        <option
          value="firstName"
        >
          First Name
        </option>
        <option
          value="lastName"
        >
          Last Name
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown snapshots matches snapshot with order toggle 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <label
      class="dropdown-label"
      for="dropdown-sort-by"
    >
      Sort by:
    </label>
    <div
      class="dropdown-control"
    >
      <select
        class="dropdown-select"
        id="dropdown-sort-by"
      >
        <option
          value="username"
        >
          Username
        </option>
        <option
          value="firstName"
        >
          First Name
        </option>
        <option
          value="lastName"
        >
          Last Name
        </option>
      </select>
      <button
        class="order-btn"
        type="button"
      >
        â†‘
      </button>
    </div>
  </div>
</DocumentFragment>
`;
